### DISCUSSION TITLE
Integracja z AZURE IoT Hub

### DISCUSSION THREAD

**Discussion by opennas** [hackerspace-silesia/Smogomierz#133] [Ideas]
Może ktoś już próbował integrować z AZURE IoT? 
Zaczynam dopiero projekt i widzę, że są biblioteki do AZURE.

**Comment by bfaliszek** [comment]
Nigdy nie korzystałem z Azure przy projektach opartych na ESP8266. Znasz może jakiś przykład takiego rozwiązania? Ten opisany tutaj jest ok? https://docs.microsoft.com/pl-pl/azure/iot-hub/iot-hub-arduino-huzzah-esp8266-get-started

**Comment by opennas** [comment]
Właśnie ten analizuję casy

https://www.thingforward.io/techblog/2018-05-22-connecting-nodemcu-to-microsoft-azure-iot-hub-part-1.html

https://bigboxofdevops.com/2018/05/09/how-azure-iot-helped-me-buy-a-new-house-part-3/

i zastanawiam się jak zintegrować Smogomierz. 

W AZURE jest ładna prezentacja wyników i można prowadzić rozbudowane analizy na większej ilości czujników w Power BI o co mi chodzi. 
http://thinglabs.io/visualize-iot-with-powerbi/ 

Mamy 5 gotowych bibliotek AZURE IoT:

1. Azure IoT library for the Arduino
https://github.com/Azure/azure-iot-arduino

2. Azure IoT Hub client for ESP8266 using MQTT
https://github.com/andriyadi/AzureIoTHubMQTTClient

3. Azure IoT HTTP Protocol library for Arduino
https://github.com/Azure/azure-iot-arduino-protocol-http

4. Azure IoT MQTT Protocol library for Arduino
https://github.com/Azure/azure-iot-arduino-protocol-mqtt 

5. Azure IoT utility library for Arduino
https://github.com/Azure/azure-iot-arduino-utility
